source/game_sa/common.cpp

```diff
@@ -314,9 +321,34 @@ void DefinedState2d() {
     RwRenderStateSet(rwRENDERSTATEALPHATESTFUNCTIONREF, (void*)2); // TODO: ?
 }
 
+// TODO: Check `outName` size (to avoid buffer overflow)
 // 0x5370A0
 void GetNameAndDamage(const char* nodeName, char* outName, bool& outDamage) {
-    plugin::Call<0x5370A0, const char*, char*, bool&>(nodeName, outName, outDamage);
+    const size_t nodesz = strlen(nodeName);
+
+    const auto TerminatedCopy = [=](size_t off) {
+        strncpy(outName, nodeName, nodesz - off);
+        outName[nodesz - off] = 0;
+    };
+
+    if (EndsWith(nodeName, "_dam", true)) {
+        outDamage = true;
+        TerminatedCopy(sizeof("_dam") - 1);
+    }
+    else {
+        outDamage = false;
+        /* Izzotop: Originally it might have looked like this
+        if (
+            nodeName[nodesz - 3] == '_' &&
+            (nodeName[nodesz - 2] == 'L' || nodeName[nodesz - 2] == 'l') &&
+            nodeName[nodesz - 1] == '0'
+        )
+        */
+        if (EndsWith(nodeName, "_l0", false))
+            TerminatedCopy(sizeof("_l0") - 1);
+        else
```

Comment:
Unfortunately ``EndsWith`` doesn't  applicable for us because ``_strnicmp`` is't cross-platform.

```suggestion
    const auto TerminatedCopy = [=](size_t offset) {
        strncpy(outName, nodeName, nodesz - offset);
        outName[nodesz - offset] = 0;
    };

    // EndsWith "_dam"
    if (nodeName[nodesz - 4] == '_' &&
        nodeName[nodesz - 3] == 'd' &&
        nodeName[nodesz - 2] == 'a' &&
        nodeName[nodesz - 1] == 'm'
    ) {
        outDamage = true;
        TerminatedCopy(sizeof("_dam") - 1);
    }
    else {
        outDamage = false;
        // EndsWith "_l0" or "_L0"
        if (
            nodeName[nodesz - 3] == '_' &&
            (nodeName[nodesz - 2] == 'L' || nodeName[nodesz - 2] == 'l') &&
            nodeName[nodesz - 1] == '0'
        ) {
            TerminatedCopy(sizeof("_l0") - 1);
        } else
```

Reply:
Nor is the code base cross platform 😅
I'd stick to `EndsWith` anyways, even without case insensitive functionality (as that would suffocate in this case as well)